awsx/ecs/containers.ts

```diff
@@ -49,7 +49,7 @@ function computeContainerDefinition(
             .apply((tgPort): schema.TaskDefinitionPortMappingInputs => {
               return {
                 containerPort: mappingInput.containerPort ?? tgPort ?? mappingInput.hostPort,
-                hostPort: tgPort ?? mappingInput.hostPort,
+                hostPort: mappingInput.hostPort ?? tgPort,
```

Comment:
@danielrbradley, is this what you had in mind for your [comment](https://github.com/pulumi/pulumi-awsx/issues/894#issuecomment-1222184866) in #894:

> Remove the implicit behaviour where we change the port number - this should highlight if there's a discrepency and fail fast.

or is there something else we should be doing to "fail fast"?

Reply:
Actually, don't think we need to throw the error. If the user's specified mismatched ports the AWS error is good enough to help them fix the bug.

I wonder if, for completeness, we should also add the container port as a fallback. E.g.:

```suggestion
                hostPort: mappingInput.hostPort ?? tgPort ?? mappingInput.containerPort,
```

Trying to think through tricky ways this could be written ...

- They just use the default port from the target group:
  - The host port is the default port.
  - We assume the container port is the same as the host (and TG default) port
- They specify the host port plus a target group:
  - The host port is now overridden and is different the the TG default
  - The container port should default to the same as the host port ... but would currently prefer the TG.
- They specify the container port and target group:
  - The container port would use its own value as expected now
  - The host port would use the target group ... this is tricky ...
    - If in awsvpc networking, this must be the same as the container port - so should prefer container over tg
    - If not in awsvpc networking, it would make more sense to use the tg (as why else did you pass it?)

Reply:
Ok, here's the full set of combinations ...

| TG default | container input | host input | container output | host output | Notes                                           |
|------------|-----------------|------------|------------------|-------------|-------------------------------------------------|
| -          | -               | -          | -                | -           | Not exposed - port mappings should be undefined |
| -          | -               | h          | h                | h           |                                                 |
| -          | c               | -          | c                | c           |                                                 |
| -          | c               | h          | c                | h           |                                                 |
| t          | -               | -          | t                | t           |                                                 |
| t          | -               | h          | h                | h           |                                                 |
| t          | c               | -          | c                | **t**       | Debatable though ... could also be `c`                 |
| t          | c               | h          | c                | h           |                                                 |